### PR TITLE
remove whitespace when m. variable referenced

### PR DESCRIPTION
**Pull request by philanderson888**

### CODE REVIEW

File: src/formatters/InteriorWhitespaceFormatter.ts [hunk 1]
```diff
@@ -83,9 +83,16 @@ export class InteriorWhitespaceFormatter {
                 continue;
             }
             isPastFirstTokenOfLine = true;
-            //force token to be exactly 1 space
             if (token.kind === TokenKind.Whitespace) {
-                token.text = ' ';
+                const isDeclaringComponentVariable = (tokens[0].kind === TokenKind.Identifier) && (tokens[0].text === 'm') && (tokens[1].kind === TokenKind.Dot);
```
Comment (TwitchBronBron) [inline src/formatters/InteriorWhitespaceFormatter.ts:87]: I'm not sure hardcoding `m` is the right approach. What about other variables? Like
- `alpha. beta = true`
- `alpha. m . beta = false`

Comment (TwitchBronBron) [reply src/formatters/InteriorWhitespaceFormatter.ts:87]: Also, what about when the variables are not the first thing in the line? Like this:
- `print m . foo`
- `doSomething(doSomethingElse(m . foo))`

**Comment by philanderson888** [comment]
In that case please specify all the unit tests that you wish the code to pass.

I was only dealing with the one specific case ie

`m.componentName = ....`   

because that is how component variables are declared in component scope, not a problem.

If you also wish me to consider other scenarios please list them

cc @TwitchBronBron @chrisdp

**Comment by TwitchBronBron** [comment]
Sure. The goal is to remove space between _any_ dotted get expressions, rather than to just the ones that start with `m`. Here are several other scenarios where the leading variable could have whitespace between the dot. 

```typescript
it('formats with optional chaining operators', () => {
    formatEqual(
        `m. alpha = m. alpha`,
        `m.alpha = m.alpha`
    );
    formatEqual(
        `m .beta = m .beta`,
        `m.beta = m.beta`
    );
    formatEqual(
        `m . charlie = m . charlie`,
        `m.charlie = m.charlie`
    );
    formatEqual(
        `alpha . beta = charlie.delta`,
        `alpha.beta = charlie.delta`
    );
    formatEqual(
        `print alpha . beta`,
        `print alpha.beta`
    );
    formatEqual(
        `doSomething(alpha . beta)`,
        `doSomething(alpha.beta)`
    );
    formatEqual(
        `doSomething(doSomethingElse( alpha . beta . charlie))`,
        `doSomething(doSomethingElse(alpha.beta.charlie))`
    );
    formatEqual(
        `value = alpha . beta + delta . charlie`,
        `value = alpha.beta + delta.charlie`
    );
    formatEqual(
        `value = 1 + (alpha . beta * (delta . charlie - 2) )`,
        `value = 1 + (alpha.beta * (delta.charlie - 2))`
    );
    formatEqual(
        `value = { alpha: beta . charlie }`,
        `value = { alpha: beta.charlie }`,
    );
});
```

**Comment by philanderson888** [comment]
Brilliant, thank you @TwitchBronBron 

I will get to it!

**Comment by philanderson888** [comment]
Hello sirs!  @TwitchBronBron @chrisdp

Today I got some fresh time to attend to this issue, and yesterday.  Combined about an extra 2 to 3 hours.

But I got a fix, all tests are now passing

Screenshot 2023-10-22 at 12 45 43

I have removed my previous code 

This now deals with everything on a purely generic basis.

I will be happy when you get a chance to review

:)

**Comment by TwitchBronBron** [comment]
@philanderson888 did you have a chance to restore this branch? I'd be happy to review it here soon if you can.

**Comment by philanderson888** [comment]
Due to the length of time that this merge request was held open I totally lost track of the fact the branch had an open merge request, and I deleted the source branch whilst doing some cleanup of my own github branches.

I had a local copy of the branch though so I was able easily to recreate all of the code changes exactly.

I have created a completely fresh fork of this branch, downloaded it then applied all code changes afresh using a different branch name.

I have pushed the new branch and the new merge request under

https://github.com/rokucommunity/brighterscript-formatter/pull/83

so I guess please follow the trail on that merge request for approval.

Thanks @TwitchBronBron

**Pull request closed without merging**